app/PaymentDrivers/Blockonomics/Blockonomics.php

```diff
@@ -132,20 +140,52 @@ public function paymentResponse(PaymentResponseRequest $request)
             'currency' => ['required'],
             'txid' => ['required'],
             'payment_method_id' => ['required'],
+            'btc_address' => ['required'],
+            'btc_amount' => ['required'],
+            'btc_price' => ['required'],
+            // Setting status to required will break the payment process
+            // because sometimes the status is returned as 0 which is falsy
+            // and the validation will fail.
+            // 'status' => ['required'],
         ]);
 
         try {
             $data = [];
-            $fiat_amount = round(($request->btc_price * $request->btc_amount), 2);
+            $fiat_amount = round(($request->btc_price * $request->btc_amount / 100000000), 2);
             $data['amount'] = $fiat_amount;
-            $data['currency'] = $request->currency;
             $data['payment_method_id'] = $request->payment_method_id;
             $data['payment_type'] = PaymentType::CRYPTO;
             $data['gateway_type_id'] = GatewayType::CRYPTO;
-            $data['transaction_reference'] = $request->txid;
 
-            $statusId = Payment::STATUS_PENDING;
+            // Append a random value to the transaction reference for test payments
+            // to prevent duplicate entries in the database.
+            // This ensures the payment hashed_id remains unique.
+            $testTxid = $this->getTestTxid();
```

Comment:
no need for the getter now, just reference the var directly

```
$this->test_txid;
```

Reply:
Good call, done!